기록 상세 화면에서 수정 버튼 기능 구현

jungkeunking opened the pull request:
- [x] 기록 상세 화면에서 수정 버튼 기능 구현
- [x] 간단한 리팩토링: 기록 저장 화면 등록 -> 저장 버튼 이름 변경 및 데이터 바인딩용 namespace를 app -> bind로 변경

ToDo?
- 수정 버튼 클릭 후 기록 저장 화면에 이동해서 지도 스냅샷에 기록에 해당하는 장소 표시
- 스트링 리소스 분리
- 오류 발생 가능성에 대해 테스트 코드 작성 및 예외 처리

** 참고사항
- url, uri 용어 정리를 팀 차원에서 한번 해야 될 거 같습니다
- 현재 구현은 홈 지도 화면 -> 기록 상세 화면 진입한 경우의 수정 버튼에만 한정되고, 피드 화면에서 진입하는 경우는 적용 안 됩니다
- 데이터바인딩 관해서 xml 파일들 좀 보다가, 프로필 화면 xml에서 안 쓰는 뷰모델이 있는데 그거 주한님이 일부러 놔두신 건지, 삭제해도 되는지 확인 부탁드립니다 (관련해서 refactor/profile 브랜치 푸시해놨습니다)
- safe args를 프래그먼트에서 관리 VS 뷰모델에서 관리 고민한 부분인데 팀원 분들 의견도 궁금합니다. 여기서는 뷰모델로 결정했습니다

-- 피드백 반영 및 추가 수정사항 --
- [x] 기록 수정시 저장 실패 오류 해결
- [x] 기록 삭제시 삭제 실패 오류 해결

-- 2차 수정사항 --
- [x] 기록 저장 화면에서 기록 수정시 새 기록 추가에서 기존 기록 수정으로 구현 오류 해결

-- 3차 수정사항 --
- [x] SaveLogViewModel 상태들을 uiState로 일괄 관리
- [x] 텍스트 입력에 따른 버튼 활성화 로직 최적화 -> 빈 칸에 입력하는 경우 혹은 빈 칸이 되는 경우만 버튼 상태 변경 함수 호출

jungkeunking committed:
remove: 임시 클래스 제거

jungkeunking committed:
feat: 기록 수정 버튼 기능 구현

jungkeunking committed:
refactor: 기록 저장 화면 등록 -> 저장 버튼 이름 변경 및 데이터 바인딩용 namespace를 app -> bind로 변경

juhan2103 commented:
@jungkeunking 역시 정근님 구현 속도가 엄청 빠르십니다! 정근님이 말씀하신 fragment_profile에서 viewModel은 제가 실수한 부분인 것 같습니다. refactor/profile에 있는 것으로 변경해도 될 것 같습니다! 

그리고 궁금한 점이 있습니다! 팝업 메뉴에서 수정을 눌렀을 때 상세 화면에 내용을 로드하는 것 까지는 확인했는데 글을 수정하고 다시 저장 버튼을 눌렀을 때 변하지 않는 것을 확인했습니다. 

정근님이 구현하신 내용은 일단 상세 화면 내용을 불러오는 기능까지만 구현하신건가요? 그 의도가 맞다면 테스트 했을 때 문제 없었습니다! 정근님 고생 많으셨습니다!!

jungkeunking commented:
> @jungkeunking 역시 정근님 구현 속도가 엄청 빠르십니다! 정근님이 말씀하신 fragment_profile에서 viewModel은 제가 실수한 부분인 것 같습니다. refactor/profile에 있는 것으로 변경해도 될 것 같습니다!
> 
> 그리고 궁금한 점이 있습니다! 팝업 메뉴에서 수정을 눌렀을 때 상세 화면에 내용을 로드하는 것 까지는 확인했는데 글을 수정하고 다시 저장 버튼을 눌렀을 때 변하지 않는 것을 확인했습니다.
> 
> 정근님이 구현하신 내용은 일단 상세 화면 내용을 불러오는 기능까지만 구현하신건가요? 그 의도가 맞다면 테스트 했을 때 문제 없었습니다! 정근님 고생 많으셨습니다!!

헉 감사합니다
테스트 엄청 꼼꼼하게 해주셨네요
전 로드된 것만 확인하고 그게 당연히 저장될 거라고 잘못 생각했습니다
고쳐보겠습니다!

jungkeunking committed:
fix: 기록 수정시 저장 실패 오류 해결

jungkeunking committed:
refactor: uri -> contentUri, url -> StorageUrl 이름 변경 및 바인딩 어답터 함수 수정

jungkeunking committed:
fix: 기록 삭제 안 되는 오류 해결

jsulim commented:
수정 구현 및 글을 수정하고 다시 저장 버튼을 눌렀을 때 변하지 않는 오류 해결하신 것까지 테스트 하였고 정상작동 확인했습니다. 고생하셨습니다!

donghyun81 commented:
기록 수정 버튼을 클릭하면 해당 기록을 표시하는 것만 구현되어 있는 건가요??
기록 수정으로 이동해서 데이터를 수정하고 저장하면 업데이트 되는 게 아니라 새로운 기록이 추가되고 있습니다.
다들 홈 화면에서 테스트해서 같은 장소에서 아이콘이 겹쳐서 추가된 걸 수정되었다고 보신 것 같습니다.
<img width="1176" alt="스크린샷 2024-03-02 오후 11 55 19" src="https://github.com/Sesac-team-03/TreasureHunt/assets/82762769/99d102b5-9aa9-4ac3-8328-ade4c0979eeb">
데이터베이스를 보시면 같은 장소에 데이터가 추가되고 있습니다.
그리고 피드화면에서 기록 상세를 통해 수정할때 홈 화면으로 이동하는데
제가 기억하기에는 수정해서 저장하면 피드화면에서 수정한 기록의 상세화면을 띄우는 걸로 기억하는데 이 부분도 맞는지 확인 바랍니다.

jungkeunking committed:
modify: LogModel에 id 프로퍼티들 추가 및 변환 함수에 반영

jungkeunking committed:
feat: LogDTO 업데이트 함수 추가

jungkeunking committed:
fix: 기록 저장 화면에서 기록 수정시 새 기록 추가에서 기존 기록 수정으로 구현 오류 해결 및 함수 리팩토링

jungkeunking commented:
> 기록 수정 버튼을 클릭하면 해당 기록을 표시하는 것만 구현되어 있는 건가요?? 기록 수정으로 이동해서 데이터를 수정하고 저장하면 업데이트 되는 게 아니라 새로운 기록이 추가되고 있습니다. 다들 홈 화면에서 테스트해서 같은 장소에서 아이콘이 겹쳐서 추가된 걸 수정되었다고 보신 것 같습니다. <img alt="스크린샷 2024-03-02 오후 11 55 19" width="1176" src="https://private-user-images.githubusercontent.com/82762769/309482958-99d102b5-9aa9-4ac3-8328-ade4c0979eeb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0NjY3NjEsIm5iZiI6MTcwOTQ2NjQ2MSwicGF0aCI6Ii84Mjc2Mjc2OS8zMDk0ODI5NTgtOTlkMTAyYjUtOWFhOS00YWMzLTgzMjgtYWRlNGMwOTc5ZWViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAzVDExNDc0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0MGVhYmI2YzFhZTQ4MDlmM2JmODhlNWRlYmIxMDEwYWI5MTNkZmY4ODYzYjY2MGY0ZjYyZWNhNjY4NTA2OGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.G82EiTQd0PfJgTJxo62Fhxnftom_QP4l0exnhAxHdIM"> 데이터베이스를 보시면 같은 장소에 데이터가 추가되고 있습니다. 그리고 피드화면에서 기록 상세를 통해 수정할때 홈 화면으로 이동하는데 제가 기억하기에는 수정해서 저장하면 피드화면에서 수정한 기록의 상세화면을 띄우는 걸로 기억하는데 이 부분도 맞는지 확인 바랍니다.

중요한 오류 찾아주셔서 감사합니다
새 데이터 추가가 아니라 기존 데이터 수정하도록 변경해서 푸시했습니다
피드화면의 경우는 PR 내용에 이미 명시했으니 참고해주세요

jungkeunking committed:
refactor: 함수 리팩토링, 이름 수정, 로그 삭제

버튼명 select photo -> pick image (뷰모델의 Intent.ACTION_PICK을 사용하는 getImagePick 함수와 통일)
프로퍼티명 isEnabled -> isSaveButtonEnabled
함수명 setImageLauncher -> addImages 등

jungkeunking committed:
feat: SaveLogViewModel 상태들을 uiState로 일괄 관리 및 텍스트 입력에 따른 버튼 활성화 로직 최적화

jungkeunking committed:
fix: 기록 수정시 텍스트 불러오는 함수 새로 추가

juhan2103 commented on app/src/main/res/layout/fragment_savelog.xml:
```diff
@@ -119,10 +119,9 @@
                 android:layout_marginHorizontal="16dp"
                 android:layout_marginTop="16dp"
-                android:onTextChanged="@{ (s, start, before, count) -> viewModel.setTextInput(s) }"
+                android:onTextChanged="@{ (s, start, before, count) -> (start == 0 || before == 0) ? viewModel.setTextFieldState(s) : null }"
```
빈 텍스트일 때만 함수 호출하는 방식 좋습니다!

juhan2103 reviewed (commented):
@jungkeunking 정근님 고생하셨습니다!
